controllers/secret_reconciler.go

```diff
@@ -85,35 +85,120 @@ func (r *SecretReconciler) Reconcile(pullSecret v1.ClusterPullSecret, namespaceN
 		} else {
 			return errors.Wrap(err, "unexpected error checking for the namespaced pull secret")
 		}
+	}
+	return nil
+}
 
-		sa := &corev1.ServiceAccount{}
-		err = r.Client.Get(ctx, client.ObjectKey{Name: "default", Namespace: namespaceName}, sa)
-		if err != nil {
-			r.Log.Info(fmt.Sprintf("error getting SA in namespace: %s, %s", namespaceName, err.Error()))
+// RemoveRef removes an PullSecret from a service account
+func (r *SecretReconciler) RemoveRef(secretName string, namespaceName string, serviceaccountName string) error {
```

Comment:
Removing references is definitely desirable and on the roadmap.

I'd much rather see a separate PR that only does that one thing, and nothing else - no refactoring of any kind.

How does that sound?